internal/app/secretless/handlers/mysql/authentication_handshake.go

```diff
@@ -0,0 +1,380 @@
+package mysql
+
+import (
+	"github.com/cyberark/secretless-broker/internal/app/secretless/handlers/mysql/protocol"
+	"github.com/cyberark/secretless-broker/internal/app/secretless/handlers/ssl"
+	"net"
+)
+
+var errNoTLS = &protocol.Error{
```

Comment:
wondering if this should be moved to the protocol with `NewGenericError`

Reply:
addressed